docs/reference/functions.md

```diff
@@ -264,3 +264,13 @@ example-value-or-externalValue:
       - externalValue
       - value
 ```
+
+## typedEnum
+
+Models described by both a `type` and an `enum` must only expose enum values that respect this type.
```

Comment:
```suggestion
When both a `type` and `enum` are defined for a property, the enum values must respect the type.
```

This makes me think this would be better off as a custom function for the oas ruleset. Afterall, this is oas/json schema specific, which means it doesn't really belong in core functions, right?